src/plugins/discover/public/application/view_components/utils/use_search.ts

```diff
@@ -111,6 +112,15 @@ export const useSearch = (services: DiscoverViewServices) => {
     requests: new RequestAdapter(),
   };
 
+  const getDatasetAutoSearchOnPageLoadPreference = () => {
+    // Checks the searchOnpageLoadPreference for the current dataset if not specifed defaults to true
+    const datasetType = data.query.queryString.getQuery().dataset?.type;
+
+    const datasetService = data.query.queryString.getDatasetService();
+
+    return datasetType ? datasetService?.getType(datasetType)?.meta?.searchOnLoad ?? true : true;
```

Comment:
Nit: `!datasetType || datasetService?.getType(datasetType)?.meta?.searchOnLoad ?? true`

Reply:
Thanks. Fixed it.